doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,812 @@
+        AppCatalog compositeCatalog{ co_await appInstaller.GetCompositeAppCatalogAsync(getCompositeAppCatalogOptions) };
+        co_await compositeCatalog.OpenAsync();
+
+        FindPackagesOptions findPackagesOptions = CreateFindOptions();
+        PackageMatchFilter filter;
+        filter.IsAdditive = true;
+        filter.Field = PackageMatchField::Id;
+        filter.Type = MatchType::Exact;
+        filter.Value = installAppId;
+        findPackagesOptions.Filters().Append(filter);
+        FindPackagesResult findPackagesResult{ co_await compositeCatalog.FindPackagesAsync(findPackagesOptions) };
+        winrt::IVectorView<ResultMatch> matches = findPackagesResult.Matches();
+        Package package = matches.GetAt(0).Package();
+
+        if (package.IsInstalling())
+        {
+            m_installPackageOperation = package.InstallProgress();
+            UpdateUIForInstall(m_installPackageOperation, uiThread, button, progressBar, statusText); 
+            // Alternatively to Cancel the operation, use:
+            // m_installPackageOperation.Cancel();
+        }
+    }
+```
+
+## 3.6. Find sources
+
+```c++ (C++ish pseudocode)
+    IAsyncOperation<AppCatalog> FindDefaultSource()
+    {
+        AppInstaller appInstaller = CreateAppInstaller();
+
+        winrt::IVectorView<AppCatalog> catalogs{ co_await appInstaller.GetAppCatalogsAsync() };
+        for (AppCatalog catalog : catalogs)
+        {
+            if (catalog.Details().Origin == AppCatalogOrigin::Default)
+            {
+                co_await catalog.OpenAsync();
+                co_return catalog;
+            }
+        }
+        winrt::throw_hresult(E_UNEXPECTED);
+    }
+```
+
+## 3.7. Open a catalog by name
+
+Open a user-added catalog by name. There is no way to use the api to add a catalog, that must be done on the command line.
+
+```c++ (C++ish pseudocode)
+
+    IAsyncOperation<AppCatalog> FindSource(std::wstring packageSource)
+    {
+        AppInstaller appInstaller = CreateAppInstaller();
+        AppCatalog catalog{ co_await appInstaller.GetAppCatalogAsync(packageSource) };
+        co_await catalog.OpenAsync();
+        co_return catalog;
+    }
+```
+
+# 4 Remarks
+
+Notes have been added inline throughout the api details.
+
+The biggest open question right now is whether the api needs to provide a way for the client to verify that the server is actually the Windows Package Manager package.
+The concern is the following scenario: 
+The client app has packageManagement capability which currently only allows install of msix applications.
+The user clicks on buttons in the client app to start an install of Microsoft.VSCode. The client app calls the server to do the install, and a UAC prompt is shown.
+The UAC prompt indicates that the app asking for permission is not Microsoft.VSCode, but rather some other app, however the user does not bother to read the prompt and simply clicks accept.

```

Comment:
Please use a reformatting tool (like Prettier) to format this document.  Check out https://github.com/microsoft/ProjectReunion/blob/main/docs/Coding-Guidelines.md#markdown for Project Reunion's guidelines.